### PR TITLE
rgw/sfs: Object Lock - Legal holds

### LINKED ISSUE #355: rgw/sfs: Object Lock - legal holds
Legal holds related implementation in the Object Lock Epic.

### PR DESCRIPTION
**Pull request by giubacc**
Object Lock - Legal holds

Extended test-sfs-object-locking.py to cover Legal Holds.
Tested Legal Hold put/removal and related object deletion activity.

Fixes: aquarist-labs/s3gw#355
Signed-off-by: Giuseppe Baccini <giuseppe.baccini@suse.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

### CODE REVIEW

File: src/rgw/driver/sfs/object.cc [hunk 1]
```diff
@@ -319,7 +319,7 @@ int SFSObject::set_obj_attrs(const DoutPrefixProvider *dpp,
                              Attrs *delattrs,
                              optional_yield y) {
   ceph_assert(objref);
-  auto meta = objref->meta;
+  auto &meta = objref->meta;
```
Comment (jecluis) [inline src/rgw/driver/sfs/object.cc:322]: these changes seem part of a different logical commit than the one described by this commit?

**Review by jecluis — CHANGES_REQUESTED** [review]
shouldn't this be two distinct logical commits? One being the test, the other about fixing obtaining an object by reference instead of copying it?

**Comment by giubacc** [comment]
> shouldn't this be two distinct logical commits? One being the test, the other about fixing obtaining an object by reference instead of copying it?

Those fixes were introduced because these specific tests failed, so I've incorporated all into one commit.
But yes, I can split this in two. :+1:

**Comment by jecluis** [comment]
Thanks. I think it then makes sense. Having the first commit of the series being the bug fixes, and the second the test enhancement, would be great IMO. :)

File: src/rgw/driver/sfs/object.cc (hunk 1 above)
Comment (giubacc) [reply src/rgw/driver/sfs/object.cc:322]: done

**Comment by giubacc** [comment]
@jecluis @irq0 can we merge this?

**Review by jecluis — APPROVED** [review]

**Pull request merged**